src/display_manager.py

```
@@ -521,6 +545,19 @@ def update_display(self):
             if self._capture_mode_active:
                 return  # Skip hardware write — content is being captured off-screen
 
+            digest = None
+            if self._dirty_tracking_enabled:
+                try:
+                    brightness = getattr(self.matrix, 'brightness', None)
+                except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)